### ISSUE TITLE
Have a look into sparse storage of the footprint data

### ISSUE THREAD

**Issue by gareth-j** [openghg/openghg#778]
Matt talked to some people recently who stored their footprints in a sparse format, can we add this type of data to the object store or implement it with the move to zarr?

**Comment by brendan-m-murphy** [comment]
There is a package called `sparse` that has N-dimensional sparse arrays, and this can be used with xarray as a "duck array".

One issue seems to be reading and writing sparse arrays to xarray: https://github.com/pydata/xarray/issues/4156. In particular, `to_netcdf` doesn't work without converting to a dense array first.

It seems like zarr has some support for this: https://github.com/zarr-developers/zarr-python/issues/152#issuecomment-447598144

One advantage of having sparse arrays would be for boundary conditions: we could have `particle_locations` and `vmr` variables with a dimension for "cardinal_direction", which could be 'n', 'e', 's', 'w'. Then to compute the modelled baseline, we just multiply these two `DataArray`s and sum, instead of having to work with the coordinates separately. (Right now this is a bad idea because each boundary condition "curtain" would have height, lat, and lon, and the particle locations would also have time, so this would be 20x the size, but for a sparse array it doesn't really take any extra space.)

**Comment by gareth-j** [comment]
Yeah it looks like there's some recent talk about getting support added to zarr: https://github.com/zarr-developers/zarr-python/issues/424

**Comment by brendan-m-murphy** [comment]
Cool, that plus the xarray duck-array functionality should make it pretty easy to use sparse arrays.

If we can't solve the storage space issue for things like time resolved footprints, we could probably have a work-around where we save all the dims, coords, and attrs, and sparse data separately, and then make a function to reconstruct the footprint. (If the zarr sparse stuff is taking a while.) Hopefully the special release will be a big enough improvement though.
